pkg/config/config.go

```diff
@@ -164,6 +132,47 @@ func LoadConfig(lokocfgPath, lokocfgVarsPath string) (*Config, hcl.Diagnostics)
 	}, nil
 }
 
+// readFiles reads all files based on the provided names and returns a map where the key is the
+// name of the file and the value is a slice of bytes holding the file's contents.
+func readFiles(names []string) (map[string][]byte, error) {
+	files := map[string][]byte{}
+
+	for _, fn := range names {
+		data, err := ioutil.ReadFile(fn) //nolint:gosec
+		if err != nil {
+			return nil, fmt.Errorf("reading file %q: %w", fn, err)
+		}
+
+		files[fn] = data
+	}
+
+	return files, nil
+}
+
+// parseHCLFiles reads the provided HCL files, merges them into a single hcl.Body and returns it.
+func parseHCLFiles(files map[string][]byte) (hcl.Body, hcl.Diagnostics) {
+	parser := hclparse.NewParser()
+
+	hclFiles := []*hcl.File{}
+
+	var diags hcl.Diagnostics
+
+	for name, data := range files {
+		hclFile, d := parser.ParseHCL(data, name)
+		if len(d) > 0 {
```

Comment:
On a fence here, as we could do either :smile: 
```suggestion
		if d.HasErrors() {
```

Or actually, perhaps you can always append. If diags are empty, nothing gets appended.

Reply:
Removed the `if`.